File: ios/sdk/src/XmppBridge.m
//
//  XmppBridge.m
//  JitsiMeet
//
//  Created by Molnár Tímea on 2020. 10. 09..
//  Copyright © 2020. cabi LLC. All rights reserved.
//
#import <Foundation/Foundation.h>

#import "XmppBridge.h"
#import "JitsiMeetEventEmitter+Private.h"

@implementation XmppBridge

static NSString *const POST_EVENT = @"org.jitsi.meet:features/xmpp-bridge#xmpp-post-method";
static NSString *const GET_EVENT = @"org.jitsi.meet:features/xmpp-bridge#xmpp-get-method";
static NSString *const UPDATE_USER_AVATAR = @"org.jitsi.meet:features/xmpp-bridge#update-user-avatar";

RCT_EXPORT_MODULE();

- (NSArray<NSString *> *)supportedEvents
{
  return @[POST_EVENT, GET_EVENT, UPDATE_USER_AVATAR];
}

- (void)registerSelfToEventEmitter {
    [[JitsiMeetEventEmitter sharedEmitter] registerXmppBridge:self];
}

- (void)callPostMethod:(NSString *)functionName
 withStringifiedParams:(NSArray *)params
            withPlugin:(NSString *)plugin {
    [self sendEvent:POST_EVENT body:@{
        @"functionName" : functionName,
        @"stringifiedParams": params,
        @"plugin": plugin == nil ? @"" : plugin
    }];
}

- (void)callGetMethod:(NSString *)functionName
 withStringifiedParams:(NSArray *)params
            withPlugin:(NSString *)plugin {
    [self sendEvent:GET_EVENT body:@{
        @"functionName" : functionName,
        @"stringifiedParams": params,
        @"plugin": plugin == nil ? @"" : plugin
    }];
}

- (void)updateUserAvatar:(NSString *_Nonnull) jsonString {
    [self sendEvent:UPDATE_USER_AVATAR body:jsonString];
}

@end

File: android/sdk/src/main/java/org/jitsi/meet/sdk/XmppBridge.java
/*
 * Copyright © 2020. cabi LLC. All rights reserved.
 */

package org.jitsi.meet.sdk;

import org.jitsi.meet.sdk.ReactInstanceManagerHolder;

public class XmppBridge {

    /**
     * Xmpp post method event.
     */
    private static final String XMPP_POST_METHOD = "org.jitsi.meet:features/xmpp-bridge#xmpp-post-method";

    /**
     * Xmpp get method event.
     */
    private static final String XMPP_GET_METHOD = "org.jitsi.meet:features/xmpp-bridge#xmpp-get-method";

    /**
     * Sends an Xmpp post method to React Native.
     */
    public static void callPostMethod(Object data) {
        ReactInstanceManagerHolder.emitEvent(XMPP_POST_METHOD, data);
    }

    /**
     * Sends an Xmpp get method to React Native.
     */
    public static void callGetMethod(Object data) {
        ReactInstanceManagerHolder.emitEvent(XMPP_GET_METHOD, data);
    }
}
